src/wp-includes/class-wp-block-type.php

```diff
@@ -365,6 +365,18 @@ public function set_props( $args ) {
 
 		$args['name'] = $this->name;
 
+		// Setup attributes if needed.
+		if ( ! is_array( $this->attributes ) ) {
+			$this->attributes = array();
+		}
```

Comment:
@peterwilsoncc, @gziolo, do you think we should make the array default value for attributes? But I think that would be a breaking change.

Reply:
Sorry, never mind this comment. I made a logical mistake here. I'm going to update the code shortly.